lua/telescope/_extensions/project.lua

```diff
@@ -0,0 +1,62 @@
+local has_telescope, telescope = pcall(require, 'telescope')
+
+if not has_telescope then
+	error('This plugins requires nvim-telescope/telescope.nvim')
+end
+
+local actions = require("telescope.actions")
+local finders = require("telescope.finders")
+local pickers = require("telescope.pickers")
+local conf = require("telescope.config").values
+local builtin = require("telescope.builtin")
+local project_dirs_file = vim.fn.stdpath('data') .. '/telescope-projects.lua'
+
+local select_project = function(opts, projects, run_task_on_selected_project)
+	pickers.new(opts, {
+		prompt_title = 'Select a project',
+		results_title = 'Projects',
+		finder = finders.new_table {results = projects},
+		sorter = conf.file_sorter(opts),
+		attach_mappings = function(prompt_bufnr)
+			local on_project_selected = function()
+				local selection = actions.get_selected_entry(prompt_bufnr)
+				actions.close(prompt_bufnr)
+				run_task_on_selected_project(selection.value)
+			end
+			actions.goto_file_selection_edit:replace(on_project_selected)
+			return true
+		end
+	}):find()
+end
+
+local project = function(opts)
+	opts = opts or {}
+
+	-- TODO figure this out:
+
+	-- local project_dirs = {}
+	-- for line in io.lines(project_dirs_file) do
+	-- 	table.insert(project_dirs, line)
+	-- end
+
+	local project_dirs = {["nvim config"] = "~/.config/nvim"}
+	local projects = {}
+
+	local search_selected_project = function(selection)
+		local selected_directory = project_dirs[selection]
+		builtin.find_files({cwd = selected_directory})
+	end
+
+	-- TODO allow a way to choose between the above function and this one.
+	local open_path_at_selected_project = function(selection)
+		vim.api.nvim_command('Explore ' .. project_dirs[selection])
+	end
```

Comment:
Most likely new picker for that one. So new function

Reply:
I'm not sure about this one @conni2461. I like the idea to pass the various functions in to do something on selection, but I wouldn't want user to have to got through 2 separate menus - 1 to choose project and then a second to choose to go to the project folder or find a file in the project. I was thinking a passed in option or a key mapping for the preference here. We could consider this for enhancement later on, but (please correct me if you think I'm wrong) I imagine 99% of the use case would be - select a project -> find a file in that project to work on.

Reply:
I am not really sure what 99% of the cases are. So it was just a idea that people might wanna switch to a new project rather than just searching in a different project once

Reply:
I obviously can't be sure either. I'm just guessing based on how I imagine people would like to use this. I know I wouldn't want to have to navigate 2 menus, personally in this instance, but yes can discuss later :-)